### PR TITLE
Fix CMessageBroker invalid JSON object processing

### PR DESCRIPTION
**Pull request by AKalinich-Luxoft**
Fixed CMessageBroker invalid JSON object processing. The root cause of the problem is that CMessageBroker drops whole message buffer when it receives some invalid json objects. It seems to be not correct because message buffer could contain also some valid json objects.

For example CMessageBroker receives buffer where:
`{valid_response_1}{invalid_response_2}{valid_response_3}...`
Then it parse JSON objects one by one until buffer become empty. `{valid_response_1}` will be parsed OK and SDL will receive parsed message, but `{invalid_response_2}` will not be parsed OK and CMessageBroker will not continue parsing objects after that. Then if CMessageBroker receives some `{new_valid_response}` and parse it OK, it will drop the whole buffer with `{invalid_response_2}{valid_response_3}...` so SDL will not receive parsed messages of `{valid_response_3}` and all others in buffer and this is a reason why SDL will not react on them.

In this commit:
- In case of invalid JSON object received CMessageBroker tries to find next JSON object after current invalid and if it was found, current invalid JSON object will be dropped and parsing will be continued
- Fixed string formatting in MB debug messages to avoid compile errors when MB logs are enabled

### CODE REVIEW

**Comment by AKalinich-Luxoft** [comment]
@AByzhynar, @LitvinenkoIra, @LuxoftAKutsan, @dev-gh please review

File: src/3rd_party-static/message_broker/src/lib_messagebroker/CMessageBroker.cpp [hunk 1]
```diff
@@ -444,26 +444,37 @@ void CMessageBroker::onMessageReceived(int fd, std::string& aJSONData, bool tryH
...
+        const bool is_object = true;
+        const size_t next_object_pos =
+            p->jumpOverJSONObjectOrArray(is_object, aJSONData);
+
+        if (next_object_pos != std::string::npos) {
+          DBG_MSG_ERROR(("Invalid JSON object probably. Skipping.\n"));
+          aJSONData.erase(0, next_object_pos);
+          DBG_MSG(("Buffer after cut is: '%s'\n", aJSONData.c_str()));
+          continue;
+        } else {
```
Comment (AByzhynar) [inline src/3rd_party-static/message_broker/src/lib_messagebroker/CMessageBroker.cpp]: @AKalinich-Luxoft Possibly `else` is redundant

Comment (AKalinich-Luxoft) [inline src/3rd_party-static/message_broker/src/lib_messagebroker/CMessageBroker.cpp]: @AByzhynar fixed in https://github.com/smartdevicelink/sdl_core/pull/1590/commits/2e33280a1c6cb077fa5a301f967c859318501e24

**Comment by theresalech** [comment]
Per Luxoft, this fixes issue #1015

**Pull request merged**